### PR TITLE
Acts as taggable / js filters

### PR DESCRIPTION
**Pull request by iAmNathanJ**
This is the first iteration of the tagging system.

### CODE REVIEW

**Comment by iAmNathanJ** [comment]
@catheraaine @BC3209 @boomingbrake Don't merge yet. Will need to populate the db and check behavior.

**Comment by iAmNathanJ** [comment]
@boomingbrake This all seems to be working properly. 😄  So we need to hide the taglist from the admin section? Then dynamically create it on every save? And make the filter controls build dynamically? We should try to work out next steps tomorrow.

File: Gemfile.lock [hunk 1]
```diff
@@ -217,8 +217,5 @@ DEPENDENCIES
   uglifier (>= 1.3.0)
   web-console (~> 2.0)
```
Comment (iAmNathanJ) [inline Gemfile.lock]: @boomingbrake I don't know why this changed.

**Comment by iAmNathanJ** [comment]
@boomingbrake @catheraaine @BC3209 Anyone want to check this out? Basic filtering should be working on the front-end.

**Comment by boomingbrake** [comment]
LOOKS GREAT!!!

**Pull request merged**